src/main/java/dev/openfeature/sdk/OpenFeatureAPI.java

```diff
@@ -76,20 +84,49 @@ public EvaluationContext getEvaluationContext() {
      * {@inheritDoc}
      */
     public void setProvider(FeatureProvider provider) {
+        if (provider == null) {
+            this.providers.remove(DEFAULT_PROVIDER_KEY);
+        } else {
+            setProvider(DEFAULT_PROVIDER_KEY, provider);
+        }
+    }
+
+    /**
+     * Add a provider for a named client.
+     * @param clientName The name of the client.
+     * @param provider The provider to set.
+     */
+    public void setProvider(String clientName, FeatureProvider provider) {
         try (AutoCloseableLock __ = providerLock.writeLockAutoCloseable()) {
-            this.provider = provider;
+            this.providers.put(clientName, provider);
         }
     }
 
     /**
      * {@inheritDoc}
      */
     public FeatureProvider getProvider() {
-        try (AutoCloseableLock __ = providerLock.readLockAutoCloseable()) {
-            return this.provider;
+        return this.providers.get(DEFAULT_PROVIDER_KEY);
+    }
+
+    /**
+     * Fetch a provider for a named client. If not found, return the default.
+     * @param name The client name to look for.
+     * @return A named {@link FeatureProvider}
+     */
+    public FeatureProvider getProviderForClientOrDefault(String name) {
```

Comment:
Personally, I think I'd rather see an overload of `getProvider`, but that's just my preference.